db/migrate/20210813103531_migrate_descriptor_fields_into_descriptors.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Descriptor fields is redundant now hashes are ordered. This migration
+# updates historical data to ensure:
+# - Hashes are sorted in the same order defined in order-fields
+# - Any keys present in the array, but absent from the hash are recorded
+# - Data missing a + in the hash key is repaired
+class MigrateDescriptorFieldsIntoDescriptors < ActiveRecord::Migration[5.2]
```

Comment:
Think this will need careful testing in training :)

Reply:
Already done several runs through of it, including some runs that were self validating. We *do* see some changes, mostly associated with the bug fix restoring missing data, but there are a few other places where we'll also show up data that was previously hidden. Mostly:
- Robot_id was previously only added to the hash, not the fields, so wouldn't show up on the summary page before, and will now.
- There are a few other cases where we have additional fields that will now show up on the batch summary page.  These seem to be cases where additional data has been added later, presumably by USG.